src/framework/analyses.ml

```diff
@@ -444,7 +444,7 @@ sig
   val paths_as_set : (D.t, G.t, C.t, V.t) ctx -> D.t list
 
   (** Returns initial state for created thread. *)
-  val threadenter : (D.t, G.t, C.t, V.t) ctx -> lval option -> varinfo -> exp list -> D.t list
+  val threadenter : ?multiple:bool -> (D.t, G.t, C.t, V.t) ctx -> lval option -> varinfo -> exp list -> D.t list
```

Comment:
The argument being optional here means that the default value has to be duplicated in all analyses, which is inconvenient and error-prone if someone doesn't. This could just be a named argument.

`ctx.spawn` can still have an optional argument. Using `ctx.spawn` is as convenient as before but the default value is just concentrated to the definition of `ctx.spawn` itself.